halo/hooks/object.cpp

```diff
@@ -119,36 +118,85 @@ extern "C" { // Demangle naming for ASM.
     DELC_OBJ_BEHAVIOR_FUNCTION_PTRS(placeholder);
     DELC_OBJ_BEHAVIOR_FUNCTION_PTRS(sound_scenery);
 
-}
+    #undef DELC_OBJ_BEHAVIOR_FUNCTION_PTRS
 
-// We're going to have to manually copy all of these, so we use another macro.
-
-#define COPY_EVENT_DEF(type, def) \
-    type ## _adjust_placement = def.adjust_placement; \
-    type ## _create = def.create; \
-    type ## _place = def.place; \
-    type ## _destroy = def.destroy; \
-    type ## _update = def.update; \
-    type ## _export_function_values = def.export_function_values; \
-    type ## _handle_deleted_object = def.handle_deleted_object; \
-    type ## _handle_region_destroyed = def.handle_region_destroyed; \
-    type ## _handle_parent_destroyed = def.handle_parent_destroyed; \
-    type ## _preprocess_node_orientations = def.preprocess_node_orientations; \
-    type ## _postprocess_node_matricies = def.postprocess_node_matricies; \
-    type ## _reset = def.reset; \
-    type ## _disconnect_from_structure_bsp = def.disconnect_from_structure_bsp; \
-    type ## _notify_impulse_sound = def.notify_impulse_sound; \
-    type ## _render_debug = def.render_debug; \
-    type ## _new_to_network = def.new_to_network; \
-    type ## _update_baseline = def.update_baseline; \
-    type ## _generate_update_header = def.generate_update_header; \
-    type ## _handle_received_update = def.handle_received_update; \
-    type ## _should_do_update_incremental = def.should_do_update_incremental; \
-    type ## _set_last_update_time = def.set_last_update_time; \
-    // End of Macro
+}
 
 static ObjectBehaviorDefinition* vanilla_def_pointers_backup[POSITIVE_OBJECT_TYPES];
 
+extern "C" {
+
+    uintptr_t after_object_hook__create;
+    extern object_create_wrapper();
+
+    uintptr_t after_weapon_hook__pull_trigger;
+    extern weapon_pull_trigger_wrapper();
+
+    uintptr_t after_biped_hook__jump;
+    extern biped_jump_wrapper();
+
+}
+
+struct ObjectCreateArgs {
+    int unknown;
+    int unknown2;
+};
+
+extern "C" bool before_object_create(ObjectCreateArgs* args){
+    return true;
+}
+
+extern "C" void after_object_create(uint32_t* obj){
```

Comment:
Why is this defined as an actual function instead of a prototype, like the other cases?

Reply:
From our conversation, this is empty because it's just not implemented yet. In the future, we should add a `// TODO implement this later` to clear up this confusion.

Reply:
Ok #45